### PR TITLE
C1774 signup changes

### PR DESCRIPTION
**Pull request by javitonino**
CartoDB/cartodb-central#1731

**Multi-part PR**
We are tracking this in the central PR: CartoDB/cartodb-central/pull/1774

### CODE REVIEW

**Comment by ivanmalagon** [comment]
Added `do not merge yet` because we need to open a new ticket only for the migration.

File: app/controllers/admin/visualizations_controller.rb [hunk 1]
```diff
@@ -257,7 +257,7 @@ def public_map
       response.headers['Cache-Control'] = "no-cache,max-age=86400,must-revalidate, public"
     end
 
-    @name = @visualization.user.name.present? ? @visualization.user.name : @visualization.user.username.truncate(20)
+    @name = @visualization.user.name_or_username
```
Comment (juanignaciosl) [inline app/controllers/admin/visualizations_controller.rb]: Shouldn't `truncate` behavior be taken into account, maybe at frontend?

Comment (javitonino) [inline app/controllers/admin/visualizations_controller.rb]: I removed it because it didn't make much sense to me. The fact is, that `Name` is a free-form text without limits, and that is not truncated. The fact that username was truncated, but the full name didn't looked very strange to me and I was feeling a little :psycho: at the time.

Also, it is already truncated at frontend where the space is tight (in the breadcrumb) and not truncated where there is more space (near the bottom):
screenshot_20170525_151354

While testing, I found a place where maybe it _should_ be truncated, but it is not too bad (it just displays a wider popup):
screenshot_20170525_151519

In summary, I think it's okay as it is.

Comment (juanignaciosl) [inline app/controllers/admin/visualizations_controller.rb]: 👍

**Comment by xavijam** [comment]
Both fields should take the same space than website field, for example.

screen shot 2017-05-26 at 12 09 10

**Comment by javitonino** [comment]
Did the backend change.

**Comment by xavijam** [comment]
Acceptance accomplished! Found just those 3 things. Taking care of the frontend ones.

File: app/assets/stylesheets/common/account_forms.css.scss [hunk 2]
```diff
@@ -280,6 +280,9 @@ $sLabel-width: 140px;
 .FormAccount-input--med {
   width: 274px !important;
 }
+.FormAccount-input--small {
+  width: 131px !important;
```
Comment (javitonino) [inline app/assets/stylesheets/common/account_forms.css.scss]: :trollface:

**Comment by xavijam** [comment]
The problem about the name in the dropdown comes from the `user_decorator`, it only includes the first name, not both, could you take care of it @javitonino ?

File: app/assets/stylesheets/common/account_forms.css.scss (hunk 2 above)
Comment (xavijam) [inline app/assets/stylesheets/common/account_forms.css.scss]: Following the same rules already described ;)
